libmamba/src/core/channel_context.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2019, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include <cassert>
+#include <tuple>
+#include <utility>
+
+#include "mamba/core/channel_context.hpp"
+#include "mamba/core/context.hpp"
+#include "mamba/specs/channel_spec.hpp"
+#include "mamba/specs/conda_url.hpp"
+#include "mamba/util/environment.hpp"
+#include "mamba/util/url_manip.hpp"
+
+namespace mamba
+{
+
+    /*********************************
+     * ChannelContext implementation *
+     *********************************/
+
+    namespace
+    {
+        auto create_paltforms(const std::vector<std::string>& platforms)
+            -> specs::ChannelResolveParams::platform_list
+        {
+            return { platforms.cbegin(), platforms.cend() };
+        }
+    }
+
+    ChannelContext::ChannelContext(Context& context)
+        : m_context(context)
+    {
+        m_channel_params.channel_alias = specs::CondaURL::parse(
+            util::path_or_url_to_url(m_context.channel_alias)
+        );
+        m_channel_params.home_dir = util::user_home_dir();
+        m_channel_params.current_working_dir = fs::current_path();
+        m_channel_params.authentication_db = m_context.authentication_info();
+        m_channel_params.platforms = create_paltforms(m_context.platforms());
```

Comment:
- These members should been initialized with a member initializer list
- typo: create_paltforms => create_platforms (I guess the declaration and the definition have the typo too, since it compiles)